### PR TITLE
Add allocations feature

### PR DESCRIPTION
**Pull request by jesseyuen**
This PR introduces the Allocations feature to the Publish prototype. 

Its quite rough around the edges code-quality wise but I've tidied it up considerably from what it looked like in the initial prototype I put together when I first started in gov. It's also not completely wired up but enough is functional  so all the essential flows are able to be viewed.

The allocations window has three states: `open`, `closed`  and `confirmed` which you can now set via config settings in `generate.rb`.

### Org landing page

localhost_3000_ (5)

### Allocations open

localhost_3000_allocations

### Allocations closed

closed

### Allocations confirmed

confirmed

### CODE REVIEW

File: app/views/allocations/check-answers.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+					<dd class="govuk-summary-list__value">
+						{{ data['requested-amount'] }}
+					</dd>
+					<dd class="govuk-summary-list__actions">
+						<a class="govuk-link" href="/allocations/request-places">
+							Change <span class="govuk-visually-hidden">places requested</span>
+						</a>
+					</dd>
+				</div>
+			</dl>
```
Comment (frankieroberto) [inline app/views/allocations/check-answers.html:41]: Use the macro here?

File: app/views/allocations/check-answers.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+		</div>	
+
+		<a class="govuk-button" href="/allocations/confirmation">Send request</a>
```
Comment (frankieroberto) [inline app/views/allocations/check-answers.html:45]:
```suggestion
    {{ govukButton({href: "/allocations/confirmation", text: "Send request"}) }}
```

File: app/views/allocations/confirmation.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      <h1 class="govuk-panel__title">
+        Request sent
+      </h1>
+      <div class="govuk-panel__body">You’ve confirmed that {{ requestingProviderName }} would like to offer PE in 2022 to 23</div>
+	  </div>
+    <h2 class="govuk-heading-m">What happens now?</h2>
+    <p class="govuk-body">You have until 10 July to change this request. After this date, we’ll send an email to (insert email address) confirming your choice and setting out next steps.</p>
+    <h2 class="govuk-heading-m">How many places will be allocated?</h2>
+    <p>In September we'll let you know how many places we've allocated for this course.</p>
+    <p>If you have any questions, contact <a href="mailto:becomingateacher@digital.education.gov.uk">becomingateacher@digital.education.gov.uk</a>.</p>
```
Comment (frankieroberto) [inline app/views/allocations/confirmation.html:30]:
```suggestion
    <p class="govuk-body">In September we'll let you know how many places we've allocated for this course.</p>
    <p class="govuk-body">If you have any questions, contact <a href="mailto:becomingateacher@digital.education.gov.uk" class="govuk-link">becomingateacher@digital.education.gov.uk</a>.</p>
```

File: app/views/allocations/request-places.html [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  
+      <form action="/allocations/check-answers" method="post" class="form">
+        <div class="govuk-form-group">
+          <fieldset class="govuk-fieldset">
+          <legend class="govuk-fieldset__legend govuk-fieldset__legend--xl govuk-!-margin-bottom-0">
+            <h1 class="govuk-fieldset__heading govuk-!-margin-bottom-6">
+              <span class="govuk-caption-xl">{{ requestingProviderName }}</span>
+              How many places would you like to request?
+            </h1>
+            <p class="govuk-body">Say how many trainees the organisation would like to recruit - make sure this number is as accurate as possible.</p>
```
Comment (frankieroberto) [inline app/views/allocations/request-places.html:27]: Drop this line? The first bit just repeats the question, and the second bit seems fairly redundant?

```suggestion
```

If you need to keep something here, it should probably use the hint text and the ARIA attributes to associate it the the label? (If you use the macro this'll get set up for you)

File: app/views/allocations/request-places.html [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+          </legend>
+          <div class="govuk-!-margin-bottom-6">
+            <label class="govuk-label" for="event-name">
+              <span class="govuk-visually-hidden">Number of places</span>
+            </label>
```
Comment (frankieroberto) [inline app/views/allocations/request-places.html:32]: Might be better to use "How many places would you like to request?" as the label, and have it also appear as the page heading?

**Review by frankieroberto — COMMENTED** [review]
I don't have much context for this, but looks good. Left a few minor suggestions.

Having tried it in a browser, I wonder whether you could drop the confirmation screens, and instead return to the index each time, perhaps with a [confirmation banner](https://design-system.service.gov.uk/components/notification-banner/#reacting-to-something-the-user-has-done) at the top instead? That avoids having to have a slightly awkward "Return to..." link at the bottom of the page.   Also doesn’t seem like it needs a huge confirmation, as you can change your request immediately anyway?

**Comment by jesseyuen** [comment]
> I wonder whether you could drop the confirmation screens, and instead return to the index each time, perhaps with a [confirmation banner](https://design-system.service.gov.uk/components/notification-banner/#reacting-to-something-the-user-has-done) at the top instead?

I take your point on this. The process is quite new research suggested users wanted as much confirmation and reassurance as possible hence the confirmation screen. It does get rather duplicitous though Im not sure there is much priority being placed on iterating this feature so I'll leave it as is.

**Review by frankieroberto — APPROVED** [review]

**Pull request merged**